designer/server/src/test/scala/pl/touk/nussknacker/ui/validation/UIProcessValidatorSpec.scala

```diff
@@ -1270,6 +1278,168 @@ class UIProcessValidatorSpec extends AnyFunSuite with Matchers with TableDrivenP
     result.warnings shouldBe ValidationWarnings.success
   }
 
+  test(
+    "validate Map service parameter based on additional config from provider - ValidationExpressionParameterValidator"
+  ) {
+    val process = createGraph(
+      List(
+        Source("inID", SourceRef(ProcessTestData.existingSourceFactory, List())),
+        Enricher(
+          "custom",
+          ServiceRef(
+            "mapParameterService",
+            List(
+              NodeParameter(
+                ParameterName("mapParam1"),
+                Expression.spel("""{'field': "valid", 'otherField': 'c'}""")
+              ),
+              NodeParameter(
+                ParameterName("mapParam2"),
+                Expression.spel("""{'field': "invalid", 'otherField': 'c'}""")
+              ),
+            )
+          ),
+          "out"
+        ),
+        Sink("out", SinkRef(ProcessTestData.existingSinkFactory, List()))
+      ),
+      List(Edge("inID", "custom", None), Edge("custom", "out", None))
+    )
+
+    val commonParamConfig = ParameterAdditionalUIConfig(
```

Comment:
thats a lot of duplicated/almost-duplicated test setup code. I get it that these tests require a lot of additional setup, but can you deduplicate it for better readability?